src/main/java/org/jenkinsci/plugins/pubsub/Message.java

```diff
@@ -205,6 +231,44 @@ public T setEventName(Enum name) {
     }
 
     /**
+     * Get the Jenkins instance URL of the master from which the event was published.
+     *
+     * @return The Jenkins instance identity.
+     * @see #getJenkinsInstanceId()
+     */
+    public String getJenkinsInstanceUrl() {
+        return get(EventProps.Jenkins.jenkins_instance_url);
+    }
+
+    /**
+     * Get the Jenkins instance identity of the master from which the event was published.
+     * <p>
+     * <strong>Note</strong> that this is not automatically added to every event since the
+     * identity key is quite large, adding a lot of weight to each event. To add the identity
+     * to all messages, simply implement a {@link MessageEnricher} and use it to call
+     * {@link #setJenkinsInstanceId()}.
+     * <p>
+     * Maybe {@link #getJenkinsInstanceUrl()} will do the trick for you
+     * in terms of working out the origin of an event.
+     * 
+     * @return The Jenkins instance identity.
+     * @see #getJenkinsInstanceUrl()
+     */
+    public String getJenkinsInstanceId() {
+        return get(EventProps.Jenkins.jenkins_instance_id);
+    }
+
+    /**
+     * Set the Jenkins instance identity of the master from which the event was published.
+     */
+    public T setJenkinsInstanceId() {
```

Comment:
Enrichers are still free to set an ID of their own choosing, including overwriting this one, but I would see that as an anti-pattern tbh. I think consumers of these events should have some confidence wrt what's going to be in these properties. In the case of this event, I'd feel strongly that they should know for sure that they will get the same value as is on the `X-instance-identity` header that they'd see in browser responses Vs some random thing that can't really be relied upon. If someone wants a different ID, then they should come up with a property name specific to whatever that is and set it on that Vs overloading this one.